### PR TITLE
+str Add mergeAll stream operator.

### PR DESCRIPTION
**Pull request by He-Pin**
References https://github.com/akka/akka/issues/19045

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala
Comment (hepin1989) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala:2905]: should this signature be curried or not @viktorklang @johanandren , or it should be ?
```scala
def mergeAll[U >: Out](eagerComplete: Boolean, those: Graph[SourceShape[U], _]*)
```

Comment (viktorklang) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala:2905]: @hepin1989 I defer to @johanandren :)

Comment (johanandren) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala:2905]: For the record, discussions going on in https://github.com/akka/akka/pull/31299#discussion_r840351828

Comment (hepin1989) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala:2905]: I will reopen them later on

**Comment by hepin1989** [comment]
@raboof @johanandren I think this one is ready to review.

**Review by johanandren — COMMENTED** [review]
Looks good

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowMergeAllSpec.scala
Comment (He-Pin) [reply akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowMergeAllSpec.scala:57]: Seems reasonable, but that should apply to `concat` etc operators too.

Comment (johanandren) [reply akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowMergeAllSpec.scala:57]: I thought we did have that, but when looking closer the only static "combiner" source factory is `Source.combine`, which takes a strategy and already covers merging I guess.

**Review by johanandren — APPROVED** [review]

**Comment by He-Pin** [comment]
I will update other PRs once this gets merged.

**Comment by He-Pin** [comment]
@patriknw @raboof ping~~

**Comment by He-Pin** [comment]
have been rebased on to current master.

**Review by johanandren — APPROVED** [review]

File: akka-stream/src/main/scala/akka/stream/javadsl/Source.scala
Comment (He-Pin) [inline akka-stream/src/main/scala/akka/stream/javadsl/Source.scala:1683]: Updated to javadsl's Source

Comment (He-Pin) [reply akka-stream/src/main/scala/akka/stream/javadsl/Source.scala:1683]: or
```those: java.util.List[_<:Graph[SourceShape[Out], _<:Any]]```?

File: akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java
Comment (He-Pin) [inline akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java:1116]: I added a Java test

**Comment by He-Pin** [comment]
@patriknw I think this one is ready for review.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**
